nucypher/utilities/networking.py

```diff
@@ -200,4 +204,62 @@ def determine_external_ip_address(
     # complete failure!
     if not rest_host:
         raise UnknownIPAddress('External IP address detection failed')
+
     return rest_host
+
+
+def _is_global_ipv4(ip: str) -> bool:
+    """Check if an IP address is a global IPv4 address according to RFC 1918."""
+    try:
+        ip = ip_address(ip.strip())
+        return isinstance(ip, IPv4Address) and ip.is_global
+    except AddressValueError:
+        return False
+
+
+def _resolve_ipv4(ip: str) -> Optional[str]:
+    """Resolve an IP address to IPv4 if required and possible."""
+    try:
+        ip = ip_address(ip.strip())
+    except AddressValueError:
+        return None
+    if isinstance(ip, IPv6Address) and ip.ipv4_mapped:
+        return str(ip.ipv4_mapped)
+    elif isinstance(ip, IPv4Address):
+        return str(ip)
```

Comment:
Seems like `ip_address(...)` returns either IPv6 or IPv4...can this `elif` go away or be changed to an `else`?

Reply:
✔️ (sort of!)

Reply:
Actually that will have a slightly different result.  This function is designed to return `None` if `not isinstance(ip, IPv6Address) and ip.ipv4_mapped`

Reply:
True., I guess return` None` if you get passed the elif then...